src/Registry.ts

```diff
@@ -129,18 +171,37 @@ export class Registry extends Evented implements Registry {
 		}
 
 		const promise = (<WidgetBaseConstructorFunction> item)();
-		this._registry.set(widgetLabel, promise);
+		this._widgetRegistry.set(label, promise);
 
 		promise.then((widgetCtor) => {
-			this._registry.set(widgetLabel, widgetCtor);
-			this.emitLoadedEvent(widgetLabel);
+			this._widgetRegistry.set(label, widgetCtor);
+			this.emitLoadedEvent(label);
 			return widgetCtor;
 		}, (error) => {
 			throw error;
 		});
 
 		return null;
 	}
+
+	public getInjector<T extends Injector>(label: RegistryLabel): T | null {
+		if (!this.hasInjector(label)) {
+			if (this.has(label)) {
+				console.warn(`Unable to find injector '${label.toString()}', did you mean to get a widget?`);
```

Comment:
i'm not convinced we need these warns, because the api is separate and returns discreet types (that extend Injector), the user thus shouldn't be able to get this wrong at design time.